frontend/packages/ux-editor/src/utils/dataModelUtils.test.ts

```diff
@@ -0,0 +1,284 @@
+    testDataModelMetadataMock[3] = {
+      ...testDataModelMetadataMock[3],
+      xsdValueType: 'SomethingElse',
+    };
+    const xsdDataType = getXsdDataTypeFromDataModelFields(
+      ComponentType.Datepicker,
+      selectedDataField,
+      testDataModelMetadataMock,
+    );
+    expect(xsdDataType).toEqual(undefined);
+  });
+});
+
+describe('getDataModelFields', () => {
+  const componentType = ComponentType.Input;
+  const bindingKey = 'simpleBinding';
+  const dataModelMetaData = dataModelMetadataMock;
+
+  it('should return data model fields when dataModelMetaData is defined', () => {
+    const dataModelFields = getDataModelFields({
+      componentType,
+      bindingKey,
+      dataModelMetaData,
+    });
+
+    expect(dataModelFields).toEqual([
+      { value: 'field1', label: 'field1' },
+      { value: 'field2', label: 'field2' },
+      { value: 'datePickerField', label: 'datePickerField' },
+    ]);
+  });
+
+  it('should return empty array when dataModelMetaData is undefined', () => {
+    const dataModelFields = getDataModelFields({
+      componentType,
+      bindingKey,
+      dataModelMetaData: undefined,
+    });
+    expect(dataModelFields).toEqual([]);
+  });
+});
+
+describe('convertDataBindingToInternalFormat', () => {
+  const testComponent = componentMocks[ComponentType.Input];
+
+  it('should return internal format when it already has internal format', () => {
+    const component = {
+      ...testComponent,
+      dataModelBindings: {
+        simpleBinding: {
+          dataType: 'dataType',
+          field: 'field',
+        },
+      },
+    };
+    const bindingKey = 'simpleBinding';
+    const internalFormat = convertDataBindingToInternalFormat(
+      component as unknown as FormItem,
+      bindingKey,
+    );
+    expect(internalFormat).toEqual({ dataType: 'dataType', field: 'field' });
+  });
+
+  it('should return correct format when it has old format', () => {
+    const bindingKey = 'simpleBinding';
+    const internalFormat = convertDataBindingToInternalFormat(testComponent, bindingKey);
+    expect(internalFormat).toEqual({ dataType: undefined, field: '' });
+  });
+
+  it('should return correct format when dataModelBindings and bindingKey is not defined', () => {
+    const component = {
+      ...testComponent,
+      dataModelBindings: undefined,
+    };
+    const bindingKey = undefined;
+    const internalFormat = convertDataBindingToInternalFormat(component, bindingKey);
+    expect(internalFormat).toEqual({ dataType: undefined, field: undefined });
+  });
+});
+
+describe('validateSelectedDataField', () => {
+  const dataModels = ['model1', 'model2'];
+
+  it('should return true when selected data model is valid', () => {
+    const selectedDataField = 'model2';
+    const isValid = validateSelectedDataModel(selectedDataField, dataModels);
+    expect(isValid).toBeTruthy();
+  });
+
+  it('should return false when selected data model is invalid', () => {
```

Comment:
Should there also be a test that checks if it returns false if selected data field is invalid/unexisting?

Reply:
Or is it jsut the test name that is mismatching here?

Reply:
in line 193 it says selectedDataField but in 194 it is the data model validator that is used 🫣

Reply:
Looking closer on the next describe block I guess it is the name of this describe block that should be changed and also the variable names in all the test cases inside it ;)